### PR TITLE
Tealv4: everything else

### PR DESCRIPTION
**Pull request by ratik21**
## Proposed Changes

+ More Versatile Global and Local Storage
+ Asset URL Change
+ Stateful Smart Contract Minimum Balance Calculation (note: *extraProgramPages* is 0 because we don't compute length)

### CODE REVIEW

File: packages/runtime/src/lib/stateful.ts [hunk 1]
```diff
@@ -3,6 +3,24 @@ import { modelsv2 } from "algosdk";
 import { RUNTIME_ERRORS } from "../errors/errors-list";
 import { RuntimeError } from "../errors/runtime-errors";
 import { StackElem } from "../types";
+import { MAX_KEY_BYTES, MAX_KEY_VAL_BYTES } from "./constants";
+
+/**
+ * Allow the combination of key-value pairs to consume 128 bytes.
```
Comment (amityadav0) [inline packages/runtime/src/lib/stateful.ts:9]: what do you mean by consume here ? could you please add more explanation here ?

File: packages/runtime/src/lib/constants.ts [hunk 2]
```diff
@@ -16,9 +16,10 @@ export const MaxTEALVersion = 4;
 // minimum balance costs (in microalgos) for ssc schema
 export const APPLICATION_BASE_FEE = 0.1e6; // base fee for creating or opt-in to application
 export const ASSET_CREATION_FEE = 0.1e6; // creation fee for asset
-export const SSC_KEY_BYTE_SLICE = 25000; // cost for 'key' (always in bytes)
-export const SSC_VALUE_UINT = 3500; // cost for value as uint64
-export const SSC_VALUE_BYTES = 25000; // cost for value as bytes
+export const SSC_VALUE_UINT = 28500; // cost for value as uint64
```
[the remark this answers is not in the document]
Comment (ratik21) [reply packages/runtime/src/lib/constants.ts:19]: yes

File: CHANGELOG.md [hunk 3]
```diff
@@ -13,6 +13,8 @@
     * Transaction Array changes
         a) array length assertions for `tx.ForeignAssets`, `tx.Accounts`, `tx.ForeignApps`,
         b) User can pass id/offset for app/asset in for array references. For `tx.Accounts` you can pass address directly in teal code.
+    * More Versatile Global and Local Storage (combination of upto 128 bytes allowed between key-value).
+    * Asset URL change (max size increased till 96 bytes).
```
[the remark this answers is not in the document]
Comment (ratik21) [reply CHANGELOG.md:18]: Technically min balance calculation remains the same (and we don't compute length).

File: packages/runtime/src/lib/stateful.ts (hunk 1 above)
Comment (ratik21) [reply packages/runtime/src/lib/stateful.ts]: done

**Pull request merged**